logging/telemetry.go

```diff
@@ -92,11 +92,12 @@ func makeTelemetryState(cfg TelemetryConfig, hookFactory hookFactory) (*telemetr
 // ReadTelemetryConfigOrDefault reads telemetry config from file or defaults if no config file found.
 func ReadTelemetryConfigOrDefault(dataDir *string, genesisID string) (cfg TelemetryConfig, err error) {
 	err = nil
-	if dataDir != nil && *dataDir != "" {
+	dataDirProvided := dataDir != nil && *dataDir != ""
+	if dataDirProvided {
 		configPath := filepath.Join(*dataDir, TelemetryConfigFilename)
 		cfg, err = LoadTelemetryConfig(configPath)
 	}
-	if err != nil && os.IsNotExist(err) {
+	if err != nil && os.IsNotExist(err) || !dataDirProvided {
```

Comment:
The change is reasonable, however, I'd like to ask for few more small changes:
* change the function parameter from `*string` to `string`. Change these on the caller as well. ( there is no concrete current use case where we attempt to pass in nil )
* in the `if err != nil && os.IsNotExist(err) || !dataDirProvided` statement, make sure to use round brackets if there is more than a single logical expression. i.e. `if (a && b) || c
* Implement a unit test for this function. Unit test should fail prior to your change and pass with your change.